modules/exampleauth/src/Auth/Source/UserClick.php

```diff
@@ -0,0 +1,216 @@
+            } catch (Exception $e) {
+                throw new Exception(sprintf(
+                    'Invalid attributes for user %s in authentication source %s: %s',
+                    $id,
+                    $this->authId,
+                    $e->getMessage(),
+                ));
+            }
+
+            $users[$id] = $attributes;
+        }
+
+        return $users;
+    }
+
+
+    /**
+     * Constructor for this authentication source.
+     *
+     * @param array $info  Information about this authentication source.
+     * @param array $config  Configuration.
+     */
+    public function __construct(array $info, array $config)
+    {
+        // Call the parent constructor first, as required by the interface
+        parent::__construct($info, $config);
+
+        if (array_key_exists('users', $config)) {
+            $users = $config['users'];
+        } else {
+            Logger::warning(
+                "Module exampleauth:UserClick misconfigured. Put users in 'users' key in your authsource.",
+            );
+
+            throw new Error\Error(Error\ErrorCodes::WRONGUSERPASS);
+        }
+
+        $this->users = $this->getUsers($users);
+    }
+
+
+    /**
+     * Initialize login.
+     *
+     * This function saves the information about the login, and redirects to a
+     * login page.
+     *
+     * @param array &$state  Information about the current authentication.
+     */
+    public function authenticate(array &$state): never
+    {
+        /*
+         * Save the identifier of this authentication source, so that we can
+         * retrieve it later. This allows us to call the login()-function on
+         * the current object.
+         */
+        $state[self::AUTHID] = $this->authId;
+
+        // Save the $state-array, so that we can restore it after a redirect
+        $id = Auth\State::saveState($state, self::STAGEID);
+
+        /*
+         * If there is only one user configured, skip the persona chooser
+         */
+        if (count($this->users) === 1) {
+            $this->handleLogin($id, 0);
+        }
+
+        /*
+         * Redirect to the login form. We include the identifier of the saved
+         * state array as a parameter to the login form.
+         */
+        $url = Module::getModuleURL('exampleauth/profileauth');
+        $params = ['AuthState' => $id];
+        $httpUtils = new Utils\HTTP();
+        $httpUtils->redirectTrustedURL($url, $params);
+    }
+
+
+    /**
+     * Attempt to log in using the given user id.
+     *
+     * On a successful login, this function should return the users attributes. On failure,
+     * it should throw an exception. If the error was caused by the user entering the wrong
+     * username or password, a \SimpleSAML\Error\Error(\SimpleSAML\Error\ErrorCodes::WRONGUSERPASS) should be thrown.
+     *
+     * Note that both the username and the password are UTF-8 encoded.
+     *
+     * @param int $id  The username the user wrote.
```

Comment:
This is not correct. I understand that we want to use an int as the key. It does mean a user cannot reuse the config from exampleauth:UserPass which uses strings as the keys.

Reply:
There are lots of leftovers from UserPass, thx for the extra attention, will clean up